### PR TITLE
Update packaging, upgrade to TypeScript 1.8

### PR DESCRIPTION
**Pull request by kitsonk**
- Migrate to npm only for dependencies
- Add typings to package.json for distribution builds
- Remove the _modules path
- Update form.ts to be compatible with TS 1.8
- Update other npm dependencies to latest versions
- Amend tslint.json to be compatible latest version

### CODE REVIEW

File: typings/tsd.d.ts [hunk 1]
```diff
@@ -1 +1,3 @@
-///<reference path="../_modules/dojo-core/typings/dojo-core/dojo-core-2.0.0-pre.d.ts"/>
+///<reference path="../node_modules/dojo-core/typings/dojo-core/dojo-core-2.0.0-alpha.2.d.ts"/>
+/// <reference path="../node_modules/dojo-loader/typings/dojo-loader/dojo-loader-2.0.0-alpha.4.d.ts" />
+/// <reference path="../tests/typings/node/node.d.ts" />
```
Comment (Tomdye) [inline typings/tsd.d.ts]: missing new line EOF

**Comment by Tomdye** [comment]
Looks good +1

File: typings/tsd.d.ts [hunk 2]
```diff
@@ -1 +1,3 @@
-///<reference path="../_modules/dojo-core/typings/dojo-core/dojo-core-2.0.0-pre.d.ts"/>
+///<reference path="../node_modules/dojo-core/typings/dojo-core/dojo-core-2.0.0-alpha.2.d.ts"/>
```
Comment (kfranqueiro) [inline typings/tsd.d.ts]: Missing space after `///`

File: src/form.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ export type FormValue = { [ key: string ]: any };
 function findFirstEnabledIndex(select: HTMLSelectElement) {
 	const length = select.options.length;
 	for (let i = 0; i < length; i++) {
-		if (!select.options[i].disabled) {
+		if (!(<HTMLOptionElement> select.options[i]).disabled) {
```
Comment (kfranqueiro) [inline src/form.ts]: I take it TS used to express `HTMLSelectElement`'s `options` as an array or collection of `HTMLOptionElement`s but 1.8 no longer does?  Seems like a let-down...

File: tslint.json [hunk 4]
```diff
@@ -17,35 +17,37 @@
...
 		"no-eval": true,
+		"no-shadowed-variable": false,
 		"no-string-literal": false,
 		"no-switch-case-fall-through": false,
-		"no-trailing-comma": true,
 		"no-trailing-whitespace": true,
 		"no-unreachable": true,
 		"no-unused-expression": false,
-		"no-unused-variable": true,
+		"no-unused-variable": false,
```
Comment (kfranqueiro) [inline tslint.json]: Why is `no-unused-variable` being set to `false`?  It's `true` in core...

File: src/form.ts (hunk 3 above)
Comment (kitsonk) [inline src/form.ts]: Actually it was an array of `any`, which is why you could have whatever you wanted as a property.  In 1.8 it was more specific to `Element`.  See Microsoft/TSJS-lib-generator#68

File: tslint.json (hunk 4 above)
Comment (kitsonk) [inline tslint.json]: You are right.  Problem in `dojo2-package-template` too.

**Comment by kitsonk** [comment]
@vansimke have you reviewed this?  If so, if you could add a comment to that effect.

**Comment by vansimke** [comment]
Sorry, haven't looked at it yet. Didn't realize this was a PR, so I didn't look for it. I'll take a look this morning.

**Comment by vansimke** [comment]
This looks good to me

File: package.json [hunk 5]
```diff
@@ -11,25 +11,25 @@
...
 	},
-	"dependencies": {
-		"dojo-core": "dojo/core"
+	"peerDependencies": {
+		"dojo-core": ">2.0.0-alpha"
 	},
 	"devDependencies": {
 		"codecov.io": "0.1.6",
-		"dojo-loader": "2.0.0-alpha.1",
+		"dojo-loader": ">2.0.0-alpha",
```
Comment (Tomdye) [inline package.json]: -beta2

Comment (kitsonk) [inline package.json]: this is still valid semver for beta.2

**Pull request merged**